echo_server.py

```diff
@@ -29,5 +33,39 @@ def start_server():
     finally:
         raise
 
+
+def response_ok(uri):
+    okay_response = "HTTP/1.1 200 OK\r\n"
+    headers = "Content-Type: text/plain\r\n"
+    body = "you requested: {}".format(uri)
+    return "{}{}{}\r\n".format(okay_response, headers, body)
```

Comment:
There is no blank line separating the headers from the body here.  This is not a well-formed HTTP response.